src/Pages/Detail/index.js

```diff
@@ -0,0 +1,258 @@
+import React, { useState, useEffect, useRef } from "react";
+import styled from "styled-components";
+import axios from "axios";
+import Aside from "./Aside";
+import Banner from "./Components/Banner";
+import Review from "./Components/Review";
+import LearningGoal from "./Components/LearningGoal";
+import WhatIsClass from "./Components/WhatIsClass";
+import Community from "./Components/Community";
+import HeadModal from "./Components/HeadModal";
+import { Hr } from "./Components/Utils";
+import { AiFillExclamationCircle } from "react-icons/ai";
+import TopHeader from "./Components/TopHeader";
+import { GetClassInfo } from "../../store/DetailReducer";
+import { useSelector, useDispatch } from "react-redux";
+import { API } from "./Components/Utils";
+
+const Detail = () => {
+  const [isTopModal, setIsTopModal] = useState(false);
+  const [scrollValue, setScrollValue] = useState(0);
+  const [selectTab, setSelectTab] = useState(0);
+  const [slickWidth, setSlickWidth] = useState(0);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const { Detail, HeaderImages } = useSelector((store) => store.DetailReducer);
+  const dispatch = useDispatch();
+  const focusTarget = useRef([]);
+  const scrollEvent = () => {
+    setScrollValue(document.documentElement.scrollTop);
+  };
+
+  useEffect(() => {
+    window.addEventListener("scroll", scrollEvent);
+    axios.get(API).then((res) => {
+      dispatch(GetClassInfo(res.data.detail));
+    });
+    return () => {
+      window.removeEventListener("scroll", scrollEvent);
+    };
+  }, []);
+
+  useEffect(() => {
+    if (scrollValue < focusTarget.current[1].offsetTop - 50) {
+      setSelectTab(1);
+    } else {
+      setSelectTab(2);
+    }
+  }, [scrollValue]);
+
+  const MoveEvent = (value) => {
+    window.scrollTo({
+      top: value - 30,
+      left: 0,
+      behavior: "smooth",
+    });
+  };
+
+  const HandleSlick = (waypoint) => {
+    if (waypoint === "right") {
+      if (slickWidth > (HeaderImages.length - 1) * -850) {
+        setSlickWidth((prev) => prev - 850);
+        setCurrentPage((prev) => prev + 1);
+      } else if ((HeaderImages.length - 1) * -850) {
+        setSlickWidth(0);
+        setCurrentPage(1);
+      }
+    }
+    if (waypoint === "left") {
+      if (0 > slickWidth) {
+        setSlickWidth((prev) => prev + 850);
+        setCurrentPage((prev) => prev - 1);
+      } else if (0 === slickWidth) {
+        setSlickWidth((HeaderImages.length - 1) * -850);
+        setCurrentPage(HeaderImages.length);
+      }
+    }
+  };
```

Comment:
```suggestion
const HandleSlick = (direction) => {
  const isFoo = slickWidth > (HeaderImages.length - 1) * -850
  const isWidthMinus = slickWidth < 0
  let result = {}
  
  if (direction === "right") {
    result = {
      width: isFoo ? (prev) => prev + 850 : 0
      page: isFoo ? (HeaderImages.length + 1) * -850 : 1
    }
  }

  if (direction === "left") {
    result = {
      width: isWidthMinus ? (prev) => prev + 850 : (HeaderImages.length - 1) * -850
      page: isWidthMinus ? (HeaderImages.length - 1) * -850 : HeaderImages.length
    }
  }
  
  setSlickWidth(result.width)
  setCurrentPage(result.page)
};

```
- 함수 소문자로 시작해주세요
- 적당한 값으로 묶으면 if 문 중첩과 setState 코드 작성을 줄일 수 있습니다